File: kawazoe/tyama_codeiq1560.php
#!/usr/bin/php
<?php

$N=10;
$F=array(3,5,7,11,13,17,19,23,29,31);
function solve($ma){
	global $N,$F;
	$i=1;
	$j;
	$n;
	$r=0;
	for(;$i<1<<$N;$i++){
		$n=0;
		$divisor=1;
		for($n=$j=0;$j<$N;$j++)if($i&(1<<$j)){$n++;$divisor*=$F[$j];}
		$items=$ma/$divisor^0;
		$r+=(($n%2*2-1)*($items*($items+1)/2)*$divisor);
	}
	return $r;
}
echo solve(100000).PHP_EOL;
echo solve(1000000000).PHP_EOL;

File: kawazoe/tyama_codeiq1560.c
#include <stdio.h>
int N=10,F[]={3,5,7,11,13,17,19,23,29,31};
long long solve(long long ma){
	int i=1,j,n;
	long long r=0;
	for(;i<1<<N;i++){
		int n=0;
		long long divisor=1;
		for(n=j=0;j<N;j++)if(i&(1<<j))n++,divisor*=F[j];
		long long items=ma/divisor;
		r+=(n%2*2-1)*items*(items+1)*divisor/2;
	}
	return r;
}
int main(){
	printf("%lld\n",solve(100000));
	printf("%lld\n",solve(1000000000));
}